packages/vehicle-rental-overlay/src/index.js

```diff
@@ -1,45 +1,84 @@
-import { divIcon } from "leaflet";
-import memoize from "lodash.memoize";
 import * as BaseMapStyled from "@opentripplanner/base-map/lib/styled";
 import { getCompaniesLabelFromNetworks } from "@opentripplanner/core-utils/lib/itinerary";
 import {
   companyType,
-  vehicleRentalMapOverlaySymbolsType,
-  stationType
+  stationType,
+  vehicleRentalMapOverlaySymbolsType
 } from "@opentripplanner/core-utils/lib/types";
 import FromToLocationPicker from "@opentripplanner/from-to-location-picker";
+import ZoomBasedMarkers from "@opentripplanner/zoom-based-markers";
 import PropTypes from "prop-types";
 import React from "react";
-import ReactDOMServer from "react-dom/server";
-import {
-  CircleMarker,
-  FeatureGroup,
-  Marker,
-  MapLayer,
-  Popup,
-  withLeaflet
-} from "react-leaflet";
-
-import { floatingBikeIcon, hubIcons } from "./bike-icons";
-import * as Styled from "./styled";
+import { FeatureGroup, MapLayer, Popup, withLeaflet } from "react-leaflet";
 
-const getStationMarkerByColor = memoize(color =>
-  divIcon({
-    className: "",
-    iconSize: [11, 16],
-    popupAnchor: [0, -6],
-    html: ReactDOMServer.renderToStaticMarkup(
-      <Styled.StationMarker color={color} />
-    )
-  })
-);
+import {
+  GenericMarker,
+  HubAndFloatingBike,
+  SharedBikeCircle
+} from "./DefaultMarkers";
 
 /**
  * This vehicle rental overlay can be used to render vehicle rentals of various
  * types. This layer can be configured to show different styles of markers at
  * different zoom levels.
  */
 class VehicleRentalOverlay extends MapLayer {
+  /**
+   * This helper method will be passed to the ZoomBasedMarkers symbolTransform prop.
+   * It creates a component that inserts a popup
+   * as a child of the specified symbol from the mapSymbols prop.
+   */
+  insertPopup = Symbol => {
+    const SymbolWrapper = ({ entity: station, zoom }) => (
+      <Symbol entity={station} zoom={zoom}>
+        {this.renderPopupForStation(
+          station,
+          station.bikesAvailable !== null && !station.isFloatingBike
+        )}
+      </Symbol>
+    );
+    SymbolWrapper.propTypes = {
+      entity: stationType.isRequired,
+      zoom: PropTypes.number.isRequired
+    };
+
+    return SymbolWrapper;
+  };
+
+  /**
+   * Builds a component to render rental vehicles
+   * based on the symbol definition.
+   * If the symbol type is a string, use the existing symbol map.
+   * If the symbol type is a component, use that instead.
+   */
+  getSymbol = mapSymbol => {
```

Comment:
This function is small enough that I think it'd be good to inline within the `convertToZoomMarkerSymbols` method.

Reply:
Addressed in https://github.com/opentripplanner/otp-ui/pull/159/commits/dcc3b3943c44cffeb34507e9127e8c3b9190098a.